go/hyperkit.go

```diff
@@ -474,24 +425,17 @@ func (h *HyperKit) buildArgs(cmdline string) {
 		a = append(a, "-U", h.UUID)
 	}
 
-	for _, p := range h.Disks {
-		// Default the driver to virtio-blk
-		driver := "virtio-blk"
-		if p.Driver != "" {
-			driver = p.Driver
-		}
-		arg := fmt.Sprintf("%d:0,%s,%s", nextSlot, driver, p.Path)
-
-		// Add on a format instruction if specified.
-		if p.Format != "" {
-			arg += ",format=" + p.Format
-		}
-		a = append(a, "-s", arg)
+	for _, disk := range h.Disks {
+		a = append(a, "-s", fmt.Sprintf("%d:0,%s", nextSlot, disk.AsArgument()))
 		nextSlot++
 	}
 
 	if h.VSock {
-		l := fmt.Sprintf("%d,virtio-sock,guest_cid=%d,path=%s", nextSlot, h.VSockGuestCID, h.StateDir)
+		path := h.VSockDir
+		if path == "" {
+			path = h.StateDir
+		}
```

Comment:
My preference would be to arrange that `h.VSockDir` is always set by the time we get to this point, by including in `execute` (I think, maybe there is somewhere better) after the sanity checks a:
```
if h.VSockDir == "" {
    h.VsockDir = h.StateDir
}
```
Probably right before `// Create files`.

Reply:
Will do.